### ISSUE TITLE
jasypt-spring-boot-starter 2.1.1 spring.profiles.active Unable to load

### ISSUE THREAD

**Issue by yonghao9531** [ulisesbocchio/jasypt-spring-boot#133]
jasypt-spring-boot-starter 2.1.1 spring.profiles.active Unable to load
bootstrap.yml 
spring:
  profiles:
    active: dev,redis
application-redis.yml
spring:
  redis:
    password: ENC(UEey4QnMF1P1VVwIWGkTug==)

RedisSystemException, Error in execution; nested exception is io.lettuce.core.RedisCommandExecutionException: NOAUTH Authentication required

**Comment by yuanjinyong** [comment]
This issue is cause by jasypt-spring-boot have conflict with spring-cloud-context-2.0.3.RELEASE.
In spring-cloud-context-2.0.3.RELEASE, have the follow change:
org\springframework\cloud\bootstrap\BootstrapApplicationListener.java
```
public void add(PropertySource<?> source) {
            if (source instanceof EnumerablePropertySource 
                    && !this.names.contains(source.getName())) {
                this.sources.addPropertySource(source);
                this.names.add(source.getName());
            }
        }
```
to
```
public void add(PropertySource<?> source) {
            // Only add map property sources added by boot, see gh-476
            if (source instanceof OriginTrackedMapPropertySource 
                    && !this.names.contains(source.getName())) {
                this.sources.addPropertySource(source);
                this.names.add(source.getName());
            }
        }
```

And after use jasypt, the "PropertySource<?> source" instance will be "com.ulisesbocchio.jasyptspringboot.wrapper.EncryptableMapPropertySourceWrapper", it is an intance of EnumerablePropertySource, but not an intance of OriginTrackedMapPropertySource.

**Comment by ulisesbocchio** [comment]
Thanks @yuanjinyong will add encryptable equivalent on next release

**Comment by ulisesbocchio** [comment]
this should be solved in 3.0.0

**Comment by yuanjinyong** [comment]
@ulisesbocchio Need update Spring cloud to Greenwich.SR3?

**Comment by WilliamShoww** [comment]
> this should be solved in 3.0.0

3.0.0 can be use with SB Hoxton.SR9 version?

**Comment by cctvckl** [comment]
has this been solved?
